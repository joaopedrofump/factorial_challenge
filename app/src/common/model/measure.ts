type Measurement = {
  name: string;
  measures: Measure[];
  mean?: {
    lastMin?: number;
    lastHour?: number;
    lastDay?: number;
  };
};

class Measure {
  value: number;
  measurement: string;
  timestamp: string;

  constructor({
    _value,
    _measurement,
    _time,
  }: {
    _value: number;
    _measurement: string;
    _time: string;
  }) {
    this.value = _value;
    this.measurement = _measurement;
    this.timestamp = _time;
  }
}

export { Measure };
export type { Measurement };
