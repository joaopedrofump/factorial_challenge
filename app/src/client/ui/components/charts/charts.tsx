import { FC } from "react";
import { Measurement } from "src/common/model/measure";
import styles from "./charts.module.css";
import { Chart } from "./components/chart/chart";

type Props = {
  measurementsList: Measurement[];
  error?: string;
  updateMeasures: (measures: Measurement) => void;
};

const Charts: FC<Props> = ({ measurementsList, error, updateMeasures }) => {
  return (
    <section className={styles.container}>
      {error && <span>{error}</span>}
      {measurementsList.map((measurement, index) => (
        <Chart
          index={index}
          key={measurement.name}
          measurement={measurement}
          updateMeasures={updateMeasures}
        />
      ))}
    </section>
  );
};

export { Charts };
