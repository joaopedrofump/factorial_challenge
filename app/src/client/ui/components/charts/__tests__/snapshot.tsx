import { render } from "@testing-library/react";
import { Charts } from "src/client/ui/components/charts/charts";
import { Measurement } from "src/common/model/measure";

jest.mock("src/client/ui/components/charts/components/chart/chart", () => ({
  Chart: () =>  <div />,
}));

describe("test several shapes of the form", () => {
  describe("when passing empty list of measures", () => {
    it("renders charts with empty chart list", () => {
      const { container } = render(
        <Charts updateMeasures={jest.fn()} measurementsList={[]} />
      );
      expect(container).toMatchSnapshot();
    });
  });
  describe("when passing one measures", () => {
    it("renders charts with one chart", () => {
      const mockedMeasures: Measurement[] = [
        {
          name: "m1",
          mean: { lastDay: 11, lastHour: 12, lastMin: 7 },
          measures: [
            { measurement: "m1", value: 12, timestamp: "t1" },
            { measurement: "m1", value: 12, timestamp: "t2" },
            { measurement: "m1", value: 12, timestamp: "t3" },
            { measurement: "m1", value: 12, timestamp: "t4" },
          ],
        },
      ];

      const { container } = render(
        <Charts updateMeasures={jest.fn()} measurementsList={mockedMeasures} />
      );
      expect(container).toMatchSnapshot();
    });
  });
  describe("when passing several measures", () => {
    it("renders charts with several charts", () => {
      const mockedMeasures: Measurement[] = [
        {
          name: "m1",
          mean: { lastDay: 11, lastHour: 12, lastMin: 7 },
          measures: [
            { measurement: "m1", value: 12, timestamp: "t1" },
            { measurement: "m1", value: 12, timestamp: "t2" },
            { measurement: "m1", value: 12, timestamp: "t3" },
            { measurement: "m1", value: 12, timestamp: "t4" },
          ],
        },
        {
          name: "m2",
          mean: { lastDay: 11, lastHour: 12, lastMin: 7 },
          measures: [
            { measurement: "m2", value: 12, timestamp: "t1" },
            { measurement: "m2", value: 12, timestamp: "t2" },
            { measurement: "m2", value: 12, timestamp: "t3" },
            { measurement: "m2", value: 12, timestamp: "t4" },
          ],
        },
        {
          name: "m3",
          mean: { lastDay: 11, lastHour: 12, lastMin: 7 },
          measures: [
            { measurement: "m3", value: 12, timestamp: "t1" },
            { measurement: "m3", value: 12, timestamp: "t2" },
            { measurement: "m3", value: 12, timestamp: "t3" },
            { measurement: "m3", value: 12, timestamp: "t4" },
          ],
        },
        {
          name: "m4",
          mean: { lastDay: 11, lastHour: 12, lastMin: 7 },
          measures: [
            { measurement: "m4", value: 12, timestamp: "t1" },
            { measurement: "m4", value: 12, timestamp: "t2" },
            { measurement: "m4", value: 12, timestamp: "t3" },
            { measurement: "m4", value: 12, timestamp: "t4" },
          ],
        },
        {
          name: "m5",
          mean: { lastDay: 11, lastHour: 12, lastMin: 7 },
          measures: [
            { measurement: "m5", value: 12, timestamp: "t1" },
            { measurement: "m5", value: 12, timestamp: "t2" },
            { measurement: "m5", value: 12, timestamp: "t3" },
            { measurement: "m5", value: 12, timestamp: "t4" },
          ],
        },
      ];

      const { container } = render(
        <Charts updateMeasures={jest.fn()} measurementsList={mockedMeasures} />
      );
      expect(container).toMatchSnapshot();
    });
  });
});
