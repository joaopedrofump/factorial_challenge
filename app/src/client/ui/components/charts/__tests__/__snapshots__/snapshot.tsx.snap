// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test several shapes of the form when passing empty list of measures renders charts with empty chart list 1`] = `
<div>
  <section
    class="container"
  />
</div>
`;

exports[`test several shapes of the form when passing one measures renders charts with one chart 1`] = `
<div>
  <section
    class="container"
  >
    <div />
  </section>
</div>
`;

exports[`test several shapes of the form when passing several measures renders charts with several charts 1`] = `
<div>
  <section
    class="container"
  >
    <div />
    <div />
    <div />
    <div />
    <div />
  </section>
</div>
`;
